### DISCUSSION TITLE
Drag and drop files into the terminal --via SFTP?

### DISCUSSION THREAD

**Discussion by henrysheehy** [sourcelair/xterm.js#4958] [Ideas]
I would be very interested in drag and dropping files into the terminal. I suppose it could be done via SFTP. Has it been done, or could anyone kindly point me in the direction as to how I could implement this?

Thanks!

**Comment by jerch** [comment]
@henrysheehy This is currently pretty involved to get done, as we dont have anything for that out of the box. What you'd need is the following (schematically):
- write some module with SFTP functionality, e.g. as terminal addon
- place a drop handler on your terminal element
- on drop, trigger the connection setup (or provide the io stream prehand), then grab the data and send it through the connection

Alternatively you can also wait for #4943, which would contain a data upload/download functionality to some degree.

**Reply by DaniAkash** [reply]
For anyone looking for drag and drop logic - I recently implemented it to my xterm.js based terminal in https://github.com/DaniAkash/agent-terminal/pull/49
